sklearn/decomposition/_pca.py

```diff
@@ -93,6 +93,7 @@ def _assess_dimension(spectrum, rank, n_samples):
                 (spectrum[i] - spectrum[j]) * (1.0 / spectrum_[j] - 1.0 / spectrum_[i])
             ) + log(n_samples)
 
+    pu = xp.asarray(pu, device=spectrum.device)
```

Comment:
`pu` is a scalar Python value (`-rank * log(2.0)` where `rank` is an `int`) so moving it to the device explicitly should not be necessary.

Reply:
If we specify the device we should probably also specify the `dtype` to avoid the automatic cast to `float64`.